### PR TITLE
Replace random_hex with uuid4

### PR DESCRIPTION
**Pull request by emord**
Found this when I realized all of our app and module ids are actually 40 characters instead of 32 like a uuid.

I can't see why we need this as uuid4.hex should be the same (except 32 instead of 40 characters) 

originally introduced in https://github.com/dimagi/dimagi-utils/commit/4208fcb2594b256d1c550040d68e0e3f7f4560e5#diff-fa41c59ed6dea1ee985593b74e785085

### CODE REVIEW

**Review by millerdev — APPROVED** [review]

**Review by dannyroberts — APPROVED** [review]
This seems fine. Will it help with anything in particular?

**Comment by emord** [comment]
@dannyroberts just less random util code. I don't think we'll ever be able to go back and make these uuid-compliant as that would be a nasty migration

**Review by mkangia — COMMENTED** [review]
and is uuid better than using random? the probably of a conflict is lower in this case?

File: corehq/apps/hqadmin/service_checks.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def check_elasticsearch():
     if cluster_health == 'red':
         return ServiceStatus(False, "Cluster health at %s" % cluster_health)
 
-    doc = {'_id': 'elasticsearch-service-check-{}'.format(random_hex()[:7]),
+    doc = {'_id': 'elasticsearch-service-check-{}'.format(uuid.uuid4().hex[:7]),
```
Comment (mkangia) [inline corehq/apps/hqadmin/service_checks.py]: i guess the chances of having two uuids with the same first 7 characters is pretty remote.

Comment (emord) [reply corehq/apps/hqadmin/service_checks.py]: should be roughly the same chance as it was before with random_hex

**Comment by emord** [comment]
> and is uuid better than using random? the probably of a conflict is lower in this case?

uuid4 is "random" https://docs.python.org/3/library/uuid.html#uuid.uuid4

**Comment by millerdev** [comment]
> and is uuid better than using random? the probably of a conflict is lower in this case?

Strictly speaking, the old way had a lower probability of collision because it had 160 bits of random data where [UUIDs only have 128](https://github.com/python/cpython/blob/d9c6564f90ead067c2e288f01825684821b7a129/Lib/uuid.py#L604) (20 bytes vs 16 bytes).

This [explanation of collision probabilities](http://preshing.com/20110504/hash-collision-probabilities/) has a table at the bottom to help put it in perspective with some real events. Unfortunately it does not list 128-bit hashes, so you can't use that table to compare 128 vs 160 bit without doing some extra math.

**Pull request merged**